applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -34,101 +35,165 @@ namespace Kratos
+            return x["reform_model_part_at_each_time_step"].GetBool();
+        } (rParameters)
+    )
+    , mVisualizationScalarVariables(
+        [&] (Parameters& x) -> std::vector<Variable<double>> {
+            x.ValidateAndAssignDefaults(GetDefaultSettings());
+            std::vector<Variable<double>> aux_list;
+            FillVariablesList<double>(x, aux_list);
+            return aux_list;
+        } (rParameters)
+    )
+    , mVisualizationVectorVariables(
+        [&] (Parameters& x) -> std::vector<Variable<array_1d<double,3>>> {
+            x.ValidateAndAssignDefaults(GetDefaultSettings());
+            std::vector<Variable<array_1d<double,3>>> aux_list;
+            FillVariablesList<array_1d<double,3>>(x, aux_list);
+            return aux_list;
+        } (rParameters)
+    )
 
-    rParameters.ValidateAndAssignDefaults(default_parameters);
-
-    // Validate if given shape functions value is admissible
-    std::set<std::string> available_shape_functions = {"standard","ausas"};
-    std::stringstream error_msg;
-
-    if (available_shape_functions.find(rParameters["shape_functions"].GetString()) == available_shape_functions.end()){
-        error_msg << "currently prescribed shape_functions : " << rParameters["shape_functions"].GetString() << std::endl;
-        error_msg << "\tAdmissible values are : standard, ausas" << std::endl;
-        KRATOS_ERROR << error_msg.str();
-    }
-
-    mShapeFunctions = rParameters["shape_functions"].GetString();
-    mReformModelPartAtEachTimeStep = rParameters["reform_model_part_at_each_time_step"].GetBool();
-
-    const unsigned int n_variables = rParameters["visualization_variables"].size();
-    for (unsigned int i_var = 0; i_var < n_variables; ++i_var){
-        std::string variable_name = rParameters["visualization_variables"].GetArrayItem(i_var).GetString();
-
-        if(KratosComponents<Variable<double>>::Has(variable_name)){
-            mVisualizationScalarVariables.push_back(KratosComponents< Variable<double> >::Get(variable_name));
-        } else if (KratosComponents< Variable< array_1d< double, 3> > >::Has(variable_name)) {
-            mVisualizationVectorVariables.push_back(KratosComponents< Variable< array_1d<double, 3> > >::Get(variable_name));
-        } else if (KratosComponents<VariableComponent<VectorComponentAdaptor< array_1d< double, 3> > > >::Has(variable_name)) {
-            mVisualizationComponentVariables.push_back(KratosComponents< VariableComponent<VectorComponentAdaptor< array_1d< double, 3> > > >::Get(variable_name));
-        } else {
-            KRATOS_ERROR << "Only double, component and vector variables are allowed in the visualization variables list." ;
-        }
-    }
+{
 }
 
-void EmbeddedSkinVisualizationProcess::ExecuteInitialize()
+EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
+    Model &rModel,
+    Parameters &rParameters)
+    : EmbeddedSkinVisualizationProcess(
+        [&] (Model& x, Parameters& y) -> ModelPart& {
```

Comment:
isn't `[&]` the default?

Reply:
If I'm not wrong, default `[]` captures nothing. `[&]` captures all symbols by reference.